app/bundles/PluginBundle/Form/Type/FeatureSettingsType.php

```diff
@@ -66,11 +66,29 @@ public function buildForm (FormBuilderInterface $builder, array $options)
 
             list ($specialInstructions, $alertType) = $integration_object->getFormNotes('leadfield_match');
 
+            /**
+             * Auto Match Integration Fields with Mautic Fields.
+             */
+            $flattenLeadFields = array();
+            foreach (array_values($leadFields) as $fieldsWithoutGroups) {
+                $flattenLeadFields = array_merge($flattenLeadFields,$fieldsWithoutGroups);
+            }
+            
+            $integrationFields  = array_keys($fields);
+            $flattenLeadFields  = array_keys($flattenLeadFields);
+            $fieldsIntersection = array_uintersect($integrationFields, $flattenLeadFields, "strcasecmp");
+
+
+            $autoMatchedFields = array();
+            foreach ($fieldsIntersection as $field) {
+              $autoMatchedFields[$field] = $field;
```

Comment:
For some reason it still doesn't work for me. I'm trying to compare the array how it should look like with the autoMatched array. If I dump `var_dump($data['leadFields'], $autoMatchedFields);`:
```
array(2) {
  ["Company"]=> string(9) "firstname"
  ["LastName"]=> string(8) "lastname"
}
array(14) {
  ["Company"]=> string(7) "Company"
  ["FirstName"]=> string(9) "FirstName"
  ["LastName"]=> string(8) "LastName"
  ["Title"]=> string(5) "Title"
  ["Email"]=> string(5) "Email"
  ["Phone"]=> string(5) "Phone"
  ["Fax"]=> string(3) "Fax"
  ["Mobile"]=> string(6) "Mobile"
  ["Website"]=> string(7) "Website"
  ["Twitter"]=> string(7) "Twitter"
  ["City"]=> string(4) "City"
  ["State"]=> string(5) "State"
  ["ZipCode"]=> string(7) "ZipCode"
  ["Country"]=> string(7) "Country"
}
```
It works for me if I change the line 84 to `$autoMatchedFields[$field] = strtolower($field);`. Do you think this could work? I'm wondering why it worked for you and not for me. Also, the line is intended only by 2 spaces instead of 4. And when I'm in it, on the line 74 is missing space after `$flattenLeadFields = array_merge($flattenLeadFields,`. :)